sacrebleu/compat.py

```diff
@@ -122,3 +122,51 @@ def sentence_chrf(hypothesis: str,
         chrf_order=order, chrf_beta=beta, chrf_whitespace=not remove_whitespace, short=False)
     metric = CHRF(args)
     return metric.sentence_score(hypothesis, references)
+
+
+def corpus_ter(hypotheses: Iterable[str],
+               references: List[Iterable[str]],
+               normalized: bool = False,
+               no_punct: bool = False,
+               asian_support: bool = False,
+               case_sensitive: bool = False) -> TERScore:
+    """
+    Computes TER on a corpus.
+
+    :param hypotheses: Stream of hypotheses.
+    :param references: Stream of references.
+    :param normalized: Enable character normalization.
+    :param no_punct: Remove punctuation.
+    :param asian_support: Enable special treatment of Asian characters.
+    :param case_sensitive: Enable case sensitivity.
+    :return: A `TERScore` object.
+    """
+    tokenizer = TOKENIZERS['tercom'](
```

Comment:
Is there a specific reason why you wouldn't pass these to the metric and let it create the tokenizer internally?

Reply:
Just to be sure, you're suggesting that the metric should accept these arguments (`normalized`, `no_punct` etc.), correct?

The constructor currently accepts `args` (which it ignores) and an optional tokenizer. Which would be preferable?

* Have `TER` accept `args` and a bunch of additional options with pre-set defaults.
* Add new CLI options and include these settings in `args` directly.

I feel that the second option, while cleaner, would lead to cascade of changes: TER tokenizer can be used with other metrics as well, so perhaps it should also be configurable from the CLI. So maybe the options should be called `--ter-tokenizer-normalize`. But then, all tokenizer classes would need to accept `args` as well and wherever they're instantiated, we would need to provide the configuration...

So maybe option 1?

Reply:
BLEU and CHRF metrics receive their own arguments through `args` and then parse them. CHRF does not have a tokenizer but BLEU has one and if you look to `BLEU.__init__()`, that's where it is initialized. It just didn't make sense to me that BLEU creates it internally whereas TER creates the tokenizer externally and passes nothing to TER as you mentioned.

I would go for option 1: Let TER consume the arguments through `args` as is done in BLEU and CHRF. The defaults can be stored in class variables as in `BLEU`. The `TER` signature as of now adds an explicit `tokenizer` argument as well, which was not done for BLEU. At some point, we'll make all these arguments explicit with default values in `__init__` also. The encapsulating through `args` was a dirty path that I choose.

Do we really want to make TER tokenizer reusable by other metrics through CLI? It is still accessible through the programming interface if one really needs it? What do you think?

Reply:
Cool, so the idea is to encapsulate the settings in `args` which (when called from `compat`) would contain keys that are not actual CLI options, correct? And when TER is instantiated from the main script, `args` would not contain these so we would fall back to defaults?

I'll remove TER from the available tokenizers, I think that makes sense.

As for the signature, if we want to only pass `args` to it, then it would also need to unpack them and figure out whether defaults were used or whether custom settings were provided (through `compat`). I think that makes it a bit less clean but I guess it's a good workaround for the time being.

Reply:
You can check the updated solution: https://github.com/mjpost/sacrebleu/pull/102/commits/2f1705f70618ec780a8298b76318d3e4981ad426